### PR TITLE
Rename fatigue to sleepiness

### LINKED ISSUE #59858: Rename fatigue and stamina to allow expansion of terms
### Is your feature request related to a problem? Please describe.

We are facing a lot of confusion over the vague term "fatigue" currently used to mean "lack of sleep". As we continue to add more different physical and mental resources that exhaust at different speed, this term cannot easily be identified and needs to be renamed.

"Stamina" is not as bad an offender, but likewise can mean many things and should be replaced with something that means exactly what we mean by this term.

In contrast, "focus", "morale", and "weariness" for example all give clear ideas what they mean and are not as easy to mistake for something else.

### Solution you would like.

1. Rename `stamina` to "breath": in the future it will be used for aerobic exercise as well as oxygen demand eg. when swimming or holding breath to move through smoke, so this name is pretty good for all purposes.
2. Rename `fatigue` to "sleepiness", abbreviated "sleep", or possibly "somnolence".

### Describe alternatives you have considered.

It's debatable if stamina needs to change or not, but I think the new term makes it easier to tell exactly what it means especially if we have other short term physical expendable resources.

### Additional context

This frees up important design space for adding new expendable physical resources. I have one in particular that will be exciting.

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Infrastructure "Rename 'fatigue' to 'sleepiness', mostly on the back end, to avoid confusing different ways of being knackered"

#### Purpose of change

Part of https://github.com/CleverRaven/Cataclysm-DDA/issues/59858, see that issue for more details. Basically, we are modeling a *lot* of things already that are different ways of exhausting physical and mental resources: stamina, fatigue, weariness, focus... it gets very confusing, and unfortunately I really want to add one more. I think before doing that, we should try to do what we can to make it harder to confuse these things for each other.

Venera is more of an expert on airways than me and doesn't want me to name 'stamina' into 'breath' so I guess let's say:

closes #59858 

#### Describe the solution

Renames "fatigue", which is very vague and describes anything from "my arms are tired" to "I need to sleep" to "that steel support is going to give way", into "sleepiness", which isn't my favourite term - it is a bit clunky - but should be hard to confuse for anything else.

This is the result of a large find-and-replace. i have tried to read through every change and make sure I didn't miss anything, but it definitely needs another look-over by someone else. I have made sure we didn't describe someone as wearing 'combat sleepiness', and in most descriptive text I've kept the word 'fatigue' unless I'm certain it's talking about need for sleep alone.

Also renamed fd_fatigue to fd_reality_tear, which may cause a json error or two, but I think is worth it for the clarity in code.

#### Describe alternatives you've considered

I considered 'drowsiness' as another option. I considered calling it just 'sleep' or 'sleepy' but they don't fit the structure we're using it in. "mod_sleepy" sounds stupid and unintuitive as a method.

#### Testing
The game compiles and runs without error

#### Additional context
I am fairly sure this is going to break something, but I haven't figured out what yet.

### CODE REVIEW

**Comment by IdleSol** [comment]
Did you forget about the documentation? Or in a separate PR?

**Comment by I-am-Erk** [comment]
Ugh. Forgot about it, thanks for the reminder

**Comment by IdleSol** [comment]
data/json/ui/fatigue.json --> data/json/ui/sleepiness.json ?

**Comment by I-am-Erk** [comment]
good catch, thanks idlesol

**Comment by Standing-Storm** [comment]
Is renaming the field `fd_fatigue` (aka Tears in Reality) to `fd_sleepiness` (plus related things like mongroups etc) intended?

**Comment by I-am-Erk** [comment]
I had assumed the field was causing fatigue, but if it's supposed to be something else we should probably give it a different name entirely.

File: doc/MAGIC.md [hunk 1]
```diff
@@ -186,9 +186,9 @@ Effect                 | Description
...
 `recharge_vehicle`     | Increases or decreases the battery charge of a vehicle or battery-connected power grid. Damage is equal to the charge (negative decreases).
-`recover_energy`       | Recovers an energy source equal to damage of the spell.  The energy source is defined in `effect_str` and may be one of `BIONIC`, `FATIGUE`, `PAIN`, `MANA` or `STAMINA`.
+`recover_energy`       | Recovers an energy source equal to damage of the spell.  The energy source is defined in `effect_str` and may be one of `BIONIC`, `SLEEPINESS`, `PAIN`, `MANA` or `STAMINA`.
 `remove_effect`        | Removes `effect_str` effects from all creatures in the aoe.
-`remove_field`         | Removes a `effect_str` field in the aoe.  Causes teleglow of varying intensity and potentially teleportation depending on field density, if the field removed is `fd_fatigue`.  (see `ter_transform` for more versatility)
+`remove_field`         | Removes a `effect_str` field in the aoe.  Causes teleglow of varying intensity and potentially teleportation depending on field density, if the field removed is `fd_sleepiness`.  (see `ter_transform` for more versatility)
```
Comment (I-am-Erk) [inline doc/MAGIC.md:191]:
```suggestion
`remove_field`         | Removes a `effect_str` field in the aoe.  Causes teleglow of varying intensity and potentially teleportation depending on field density, if the field removed is `fd_reality_tear`.  (see `ter_transform` for more versatility)
```

**Pull request merged**